frontend/components/HeaderNav.tsx

```diff
@@ -0,0 +1,64 @@
+import React, { Component } from 'react';
+import { StyleSheet } from 'react-native';
+import { Appbar, Searchbar } from 'react-native-paper';
+import StoreSelector from './StoreSelectorButton';
+import { headerStates } from './States';
+
+export interface Props {
+    currState: string;
+}
+
+export class HeaderNav extends Component {
+    
+    constructor(props : Props) {
+        super(props); 
+        let currState = headerStates.title;
+        if (props != null && props.currState) {
```

Comment:
Did you mean to use 2 equal signs here? If so, it would probably be equivalent to write `if (props && props.currState)`, which is a pretty common pattern in JavaScript, since # any value of `props` that is neither null not undefined will be "truthy"